### PR TITLE
dbplyr 2.2.0 compat

### LINKED ISSUE #492: Unable to get dplyr to work with bigrquery
I am unable to get `dplyr` to work with `bigrquery`. The `DBI` interface works well, but when I try to use `dplyr::tbl`, I get an error. Running `show_query()` on the `tbl` suggests that the dataset name is not prepended to the table name. Wondering if that might be the issue.

```r
# Load packages
library(bigrquery)
library(dplyr)

# Connect to bigquery
bq_auth(path = Sys.getenv("SERVICE_TOKEN_JSON"))
con <- dbConnect(
  bigquery(),
  project = "publicdata",
  dataset = "samples",
  billing = Sys.getenv("BILLING_PROJECT_ID")
)

DBI::dbReadTable(con, 'natality', n_max = 5)
#> # A tibble: 5 × 31
#>   source_year  year month   day  wday state is_male child_race weight_pounds
#>         <int> <int> <int> <int> <int> <chr> <lgl>        <int>         <dbl>
#> 1        1969  1969    11    29    NA KS    TRUE             1          6.00
#> 2        1969  1969     2     6    NA MN    FALSE            1          8.94
#> 3        1970  1970     9     4    NA LA    TRUE             1          7.13
#> 4        1970  1970     1    24    NA NC    FALSE            1          7.63
#> 5        1970  1970     6     6    NA SC    TRUE             2          9.00
#> # … with 22 more variables: plurality <int>, apgar_1min <int>,
#> #   apgar_5min <int>, mother_residence_state <chr>, mother_race <int>,
#> #   mother_age <int>, gestation_weeks <int>, lmp <chr>, mother_married <lgl>,
#> #   mother_birth_state <chr>, cigarette_use <lgl>, cigarettes_per_day <int>,
#> #   alcohol_use <lgl>, drinks_per_week <int>, weight_gain_pounds <int>,
#> #   born_alive_alive <int>, born_alive_dead <int>, born_dead <int>,
#> #   ever_born <int>, father_race <int>, father_age <int>, record_weight <int>

tbl(con, 'natality') %>% 
  head()
#> Warning: <BigQueryConnection> uses an old dbplyr interface
#> ℹ Please install a newer version of the package or contact the maintainer
#> This warning is displayed once every 8 hours.
#> Error in UseMethod("op_can_download"): no applicable method for 'op_can_download' applied to an object of class "NULL"
```

### LINKED ISSUE #495: biqrquery not compatible with dbplyr 2.2.0
getting issues with `dbplyr` 2.2.0 having to use `dbplyr` 2.1.1 as work around. Unable to currently provide reprex as example but has also been noticed in  #492.  

```
Warning message:
“<BigQueryConnection> uses an old dbplyr interface
ℹ Please install a newer version of the package or contact the maintainer
This warning is displayed once every 8 hours.”
Applying predicate on the first 100 rows
Error in UseMethod("op_can_download"): no applicable method for 'op_can_download' applied to an object of class "NULL"
```

### PR DESCRIPTION
**Pull request by mgirlich**
Fixes #495 and fixes #492.
Note that you still get the warning about the old interface. It would be great if you could also switch to the [2nd edition](https://dbplyr.tidyverse.org/articles/backend-2.html). The old interface is deprecated and we plan to remove it next year.

### CODE REVIEW

**Comment by kieran-mace** [comment]
Looking forward to getting this PR approved and merged! Anything I can do to help?

**Comment by mgirlich** [comment]
@jennybc I saw you are one of the authors of bigrquery. Maybe you want to review and merge this? I think this would make people very happy 😄

File: R/dplyr.R [hunk 1]
```diff
@@ -139,23 +141,57 @@ collect.tbl_BigQueryConnection <- function(x, ...,
...
+  if (!rlang::is_empty(x$where)) return(FALSE)
+  if (!rlang::is_empty(x$order_by)) return(FALSE)
+  if (!rlang::is_false(x$distinct)) return(FALSE)
+
+  TRUE
+}
+
+is_select_trivial <- function(select, vars_prev) {
+  identical(select$name, vars_prev) &&
+    purrr::every(select$expr, rlang::is_symbol) &&
```
Comment (jennybc) [inline R/dplyr.R:169]: I don't believe purrr is currently used by bigrquery.

File: tests/testthat/test-dplyr.R [hunk 2]
```diff
@@ -48,25 +48,6 @@ test_that("can collect and compute (no dataset)", {
...
-  temp <- dplyr::compute(bq_mtcars)
-
-  # compute: persistent
-  name <- paste0("basedata.", random_name())
-  temp <- dplyr::compute(bq_mtcars, temporary = FALSE, name = name)
-  on.exit(DBI::dbRemoveTable(con, name))
-
-  expect_true(DBI::dbExistsTable(con, name))
-})
-
```
Comment (jennybc) [inline tests/testthat/test-dplyr.R:69]: What's up with removing this test?

**Review by jennybc — COMMENTED** [review]
The bigrquery <--> dbplyr interface is outside of my wheelhouse, so I'm relying on you @mgirlich re: content. But I've made some comments, as a second set of 👀, and I approved the GHA checks.

File: tests/testthat/test-dplyr.R (hunk 2 above)
Comment (jennybc) [reply tests/testthat/test-dplyr.R:69]: Now I see in a commit message that this is a duplicate test.

File: R/dplyr.R (hunk 1 above)
Comment (jennybc) [reply R/dplyr.R:169]: Yeah, GHA shows us that this is a problem.

**Comment by mgirlich** [comment]
Sorry for the late reply. Completely forgot about this... Now this does not rely on purrr anymore and should hopefully work.

**Comment by jennybc** [comment]
Thanks @mgirlich!

**Comment by jennybc** [comment]
@mgirlich bigrquery is now failing CI on the `main` branch. In hindsight, even though I approved GHA for this PR, I guess the full live API workflow did not run.

Can you have a look at this failure and make a new PR?

https://github.com/r-dbi/bigrquery/runs/8094638890?check_suite_focus=true#step:9:132

**Comment by aarizvi** [comment]
Hi all - I'm not sure if this issue is actually fixed. Here is an example from a public dataset.

```
> library(bigrquery)
> library(tidyverse)
── Attaching packages ─────────────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.8     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1
── Conflicts ────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
> 
> conn <- dbConnect(
+   bigquery(),
+   project = "isb-cgc-bq",
+   dataset = "TCGA_versioned"
+ )
> 
> conn |>
+   DBI::dbListTables() |>
+   head()
[1] "DNA_methylation_chr10_hg19_gdc_2017_01" "DNA_methylation_chr10_hg38_gdc_2017_01"
[3] "DNA_methylation_chr11_hg19_gdc_2017_01" "DNA_methylation_chr11_hg38_gdc_2017_01"
[5] "DNA_methylation_chr12_hg19_gdc_2017_01" "DNA_methylation_chr12_hg38_gdc_2017_01"
> 
> conn |> 
+   tbl("DNA_methylation_chr10_hg19_gdc_2017_01")
Error in UseMethod("op_can_download") : 
  no applicable method for 'op_can_download' applied to an object of class "NULL"
In addition: Warning message:
<BigQueryConnection> uses an old dbplyr interface
ℹ Please install a newer version of the package or contact the maintainer
This warning is displayed once every 8 hours. 
```

```
> sessionInfo()
R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.3.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.9     purrr_0.3.4     readr_2.1.2    
 [6] tidyr_1.2.0     tibble_3.1.8    ggplot2_3.3.6   tidyverse_1.3.2 bigrquery_1.4.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9          cellranger_1.1.0    pillar_1.8.0        compiler_4.2.1     
 [5] dbplyr_2.2.1        tools_4.2.1         bit_4.0.4           lubridate_1.8.0    
 [9] googledrive_2.0.0   jsonlite_1.8.0      lifecycle_1.0.1     gargle_1.2.0       
[13] gtable_0.3.0        pkgconfig_2.0.3     rlang_1.0.4         reprex_2.0.1       
[17] DBI_1.1.3           cli_3.3.0           rstudioapi_0.13     curl_4.3.2         
[21] haven_2.5.0         xml2_1.3.3          withr_2.5.0         httr_1.4.3         
[25] hms_1.1.1           generics_0.1.3      fs_1.5.2            vctrs_0.4.1        
[29] googlesheets4_1.0.0 bit64_4.0.5         grid_4.2.1          tidyselect_1.1.2   
[33] glue_1.6.2          R6_2.5.1            fansi_1.0.3         readxl_1.4.0       
[37] tzdb_0.3.0          modelr_0.1.8        magrittr_2.0.3      ellipsis_0.3.2     
[41] backports_1.4.1     scales_1.2.0        rvest_1.0.2         assertthat_0.2.1   
[45] colorspace_2.0-3    utf8_1.2.2          stringi_1.7.8       munsell_0.5.0      
[49] broom_1.0.0         crayon_1.5.1   
```

**Comment by aarizvi** [comment]
Reverting back to dbplyr 2.1.1

```
remove.packages('dbplyr')
devtools::install_version("dbplyr", version="2.1.1")
library(bigrquery)
library(tidyverse)
conn <- dbConnect(
  bigquery(),
  project = "isb-cgc-bq",
  dataset = "TCGA_versioned"
)

conn |>
  DBI::dbListTables() |>
  head()

conn |> 
  tbl("DNA_methylation_chr10_hg19_gdc_2017_01")

# Source:   table<DNA_methylation_chr10_hg19_gdc_2017_01> [?? x 7]
# Database: BigQueryConnection
   project_short_name case_barcode sample_barcode   aliquot_ba…¹ probe…² beta_…³ platf…⁴
   <chr>              <chr>        <chr>            <chr>        <chr>     <dbl> <chr>  
 1 TCGA-OV            TCGA-01-0628 TCGA-01-0628-11A TCGA-01-062… cg0376…    0.01 HumanM…
 2 TCGA-OV            TCGA-01-0630 TCGA-01-0630-11A TCGA-01-063… cg0283…    0.64 HumanM…
 3 TCGA-OV            TCGA-01-0630 TCGA-01-0630-11A TCGA-01-063… cg2627…    0.01 HumanM…
 4 TCGA-OV            TCGA-01-0630 TCGA-01-0630-11A TCGA-01-063… cg0237…    0.83 HumanM…
 5 TCGA-OV            TCGA-01-0630 TCGA-01-0630-11A TCGA-01-063… cg2176…    0.45 HumanM…
 6 TCGA-OV            TCGA-01-0631 TCGA-01-0631-11A TCGA-01-063… cg2466…    0.01 HumanM…
 7 TCGA-OV            TCGA-01-0631 TCGA-01-0631-11A TCGA-01-063… cg0087…    0.02 HumanM…
 8 TCGA-OV            TCGA-01-0631 TCGA-01-0631-11A TCGA-01-063… cg1600…    0.88 HumanM…
 9 TCGA-OV            TCGA-01-0631 TCGA-01-0631-11A TCGA-01-063… cg1301…    0.82 HumanM…
10 TCGA-OV            TCGA-01-0633 TCGA-01-0633-11A TCGA-01-063… cg2485…    0.01 HumanM…
# … with more rows, and abbreviated variable names ¹​aliquot_barcode, ²​probe_id,
#   ³​beta_value, ⁴​platform
# ℹ Use `print(n = ...)` to see more rows
```

**Comment by rappster** [comment]
I can confirm the same issue as @aarizvi with `dbplyr 2.2.1`

**Comment by aarizvi** [comment]
I see. I hadn't installed the dev version from on GitHub. That seems to work.

```
> library(bigrquery)
> library(tidyverse)
> conn <- dbConnect(
+   bigquery(),
+   project = "isb-cgc-bq",
+   dataset = "TCGA_versioned"
+ )
> 
> conn |>
+   DBI::dbListTables() |>
+   head()
Auto-refreshing stale OAuth token.
[1] "DNA_methylation_chr10_hg19_gdc_2017_01" "DNA_methylation_chr10_hg38_gdc_2017_01"
[3] "DNA_methylation_chr11_hg19_gdc_2017_01" "DNA_methylation_chr11_hg38_gdc_2017_01"
[5] "DNA_methylation_chr12_hg19_gdc_2017_01" "DNA_methylation_chr12_hg38_gdc_2017_01"
> 
> conn |> 
+   tbl("DNA_methylation_chr10_hg19_gdc_2017_01")
# Source:   table<DNA_methylation_chr10_hg19_gdc_2017_01> [?? x 7]
# Database: BigQueryConnection
   project_short_name case_barcode sample_barcode   aliquot_ba…¹ probe…² beta_…³ platf…⁴
   <chr>              <chr>        <chr>            <chr>        <chr>     <dbl> <chr>  
 1 TCGA-OV            TCGA-01-0628 TCGA-01-0628-11A TCGA-01-062… cg0376…    0.01 HumanM…
 2 TCGA-OV            TCGA-01-0630 TCGA-01-0630-11A TCGA-01-063… cg0283…    0.64 HumanM…
 3 TCGA-OV            TCGA-01-0630 TCGA-01-0630-11A TCGA-01-063… cg2627…    0.01 HumanM…
 4 TCGA-OV            TCGA-01-0630 TCGA-01-0630-11A TCGA-01-063… cg0237…    0.83 HumanM…
 5 TCGA-OV            TCGA-01-0630 TCGA-01-0630-11A TCGA-01-063… cg2176…    0.45 HumanM…
 6 TCGA-OV            TCGA-01-0631 TCGA-01-0631-11A TCGA-01-063… cg2466…    0.01 HumanM…
 7 TCGA-OV            TCGA-01-0631 TCGA-01-0631-11A TCGA-01-063… cg0087…    0.02 HumanM…
 8 TCGA-OV            TCGA-01-0631 TCGA-01-0631-11A TCGA-01-063… cg1600…    0.88 HumanM…
 9 TCGA-OV            TCGA-01-0631 TCGA-01-0631-11A TCGA-01-063… cg1301…    0.82 HumanM…
10 TCGA-OV            TCGA-01-0633 TCGA-01-0633-11A TCGA-01-063… cg2485…    0.01 HumanM…
# … with more rows, and abbreviated variable names ¹​aliquot_barcode, ²​probe_id,
#   ³​beta_value, ⁴​platform
# ℹ Use `print(n = ...)` to see more rows
> packageVersion("dbplyr")
[1] ‘2.2.1.9000’
```

**Comment by mgirlich** [comment]
> I can confirm @aarizvi observations:
> 
> * same issue for `dbplyr 2.2.1`
> * works fine for `dbplyr 2.1.1`
> * getting a new error for `dbplyr 2.2.1.9000`
> 
> ```
> r$> data <- dplyr::tbl(con, from = "<table-name>")
> Warning message:
> <BigQueryConnection> uses an old dbplyr interface
> ℹ Please install a newer version of the package or contact the maintainer
> This warning is displayed once every 8 hours. 
> ```

This is not an error but a warning (which I mentioned as not fixed in the first comment). The warning might be a bit annoying but it only appears every 8 hours and you should still be able to use the package 😄

**Comment by ari-kotler** [comment]
@mgirlich Hi there, my apologies for reviving this thread. I'm getting the following error even when using `dbplyr 2.2.1.9000`: 

`Error in UseMethod("db_analyze") : 
  no applicable method for 'db_analyze' applied to an object of class "c('tbl_BigQueryConnection', 'tbl_dbi', 'tbl_sql', 'tbl_lazy', 'tbl')"`

**Comment by mgirlich** [comment]
Sounds to me like you're not using the dev version of bigrquery. Install it via

```r
# install.packages('devtools')
devtools::install_github("r-dbi/bigrquery")
```

and restart your session.

**Comment by aarizvi** [comment]
> Hmmm I have both the dev version of dbplyr and bigrquery and still no joy on this

Unfortunately we are too. We have had to revert our pipelines to dbplyr 2.1.1 for usability.

**Comment by mgirlich** [comment]
@mvellinger @aarizvi @ari-kotler Can you install

* dbplyr 2.2.1
* dev version of bigrquery via `install_github("r-dbi/bigrquery#512")`

(remember to restart your R session before!) and give feedback whether this fixes your issues. If not, please open a new issue and tag me.

**Comment by eacatalant** [comment]
In case it helps someone else -- I was following these version instructions and still having the problem until I changed my package load order to this:?
`library(dplyr)
library(dbplyr)
library(bigrquery) `

**Pull request merged**